src/mem/AllocatorProxy.h

```diff
@@ -9,9 +9,10 @@
 #ifndef _SQUID_SRC_MEM_ALLOCATORPROXY_H
 #define _SQUID_SRC_MEM_ALLOCATORPROXY_H
 
+#include "mem/Meter.h"
+
```

Comment:
While AllocatorProxy.h does not need the include for itself, all .cc files using AllocatorProxy.h must have Meter.h included in order to use it. This one seemingly unnecessary line is avoiding a large number of .cc #include's

Reply:
> While AllocatorProxy.h does not need the include for itself, all .cc files using AllocatorProxy.h must have Meter.h included in order to use MEMPROXY_CLASS macro.

`.cc` files do not need Meter.h just because their classes are MEMPROXY_CLASSes. And even if the above assertion were true, it is irrelevant, because `MEMPROXY_CLASS` is primarily used in Foo.h header files, and those Foo.h files are usually used by many other files, not just the corresponding Foo.cc files.

There are use cases where a directly-unused `#include` is a good idea. Those cases are quite rare. This is not one of those exceptional cases AFAICT.

> This one seemingly unnecessary line is avoiding a large number of .cc #include's

I believe the official code (that does not have this line) proves that the lack of Meter declaration in this header file does not trigger injection of a large number of `#include` statements in .cc files. Have you actually tried the suggested changes?

Reply:
Done in [19748ac](https://github.com/squid-cache/squid/pull/1183/commits/19748aca94cc396d4397ef11664d8cb0bd75d6cf).